### PR TITLE
Fixes #18228 - Add CV/CCV info to client facts

### PR DESCRIPTION
**Pull request by jcpunk**
This adds some structured facts to the client YAML dump which will resolve Feature #18228.

These facts should be sufficient for auditing the state of a content view within a puppet run as well as performing an audit of composite view components.

### CODE REVIEW

**Comment by jcpunk** [comment]
Fixed style guidelines

**Comment by jlsherrill** [comment]
[test]

**Comment by jcpunk** [comment]
hmmm, I never can seem to find the reason why things failed in Jenkins....  little help?

**Comment by jlsherrill** [comment]
Yeah, its a fine art :)

checkstyle failed with:
```
00:18:40.316 
00:18:40.316 /var/lib/workspace/workspace/test_katello_pull_request/slave/fast/plugin/app/models/katello/concerns/host_managed_extensions.rb:49:7: C: Assignment Branch Condition size for info_with_katello is too high. [72.06/63]
00:18:40.316       def info_with_katello
00:18:40.316       ^^^
```

Although there aren't a lot of 'branch conditions' unless its counting all the .try() methods.  Let me look a little closer

**Comment by jlsherrill** [comment]
Looks like the .try()'s are affecting this. I think here its perfectly okay to have a bunch of those, i'd just try
adding `      #rubocop:disable Metrics/AbcSize` right before the method, like: https://github.com/Katello/katello/blob/master/app/models/setting/content.rb#L3

File: app/models/katello/concerns/host_managed_extensions.rb [hunk 1]
```diff
@@ -47,12 +47,31 @@ def smart_proxy_ids_with_katello
       end
 
       def info_with_katello
+        #rubocop:disable Metrics/AbcSize
```
Comment (jlsherrill) [inline app/models/katello/concerns/host_managed_extensions.rb]: needs to go above the method definition

**Comment by jcpunk** [comment]
Added and re-pushed.  Hmmm seems hound doesn't like that...

File: app/models/katello/concerns/host_managed_extensions.rb (hunk 1 above)
Comment (jlsherrill) [inline app/models/katello/concerns/host_managed_extensions.rb]: needs to go above the method definition, not below

Comment (jcpunk) [inline app/models/katello/concerns/host_managed_extensions.rb]: ahh, thanks!

**Comment by jcpunk** [comment]
I believe I've got it now.

**Comment by jlsherrill** [comment]
[test]

**Comment by jcpunk** [comment]
I think I cleaned up the test errors with some more `.try()` blocks

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
ACK thanks @jcpunk

**Pull request merged**